test/python/qasm3/test_export.py

```diff
@@ -429,11 +422,14 @@ def test_custom_gate_with_params_bound_main_call(self):
 
         circuit.assign_parameters({parameter0: pi, parameter1: pi / 2}, inplace=True)
 
+        # NOTE: this isn't exactly what we want; note that the parameters in the signature are not
+        # actually used.  It would be fine to change the output of the exporter to make `custom` non
+        # parametric in this case.
```

Comment:
(just to make sure I am following) This isn't what we want because we want the parameters in the signature to be used in the definition instead of `pi`/`pi/4`? Where would the parameter value be stored then?

I can see that at least the issue from #7335 (the value being copied over) isn't there anymore.